### ISSUE TITLE
Strange behavior with non 100% browser zoom

### ISSUE THREAD

**Issue by JSibir** [FNNDSC/ami#255]
Got **StackHelper** and **LocalizerHelper** strange behavior when loading AMI with different browser zoom.

In my code I have this (taken from examples):
```
...stackHelper.slice.canvasWidth = ...clientWidth;
...stackHelper.slice.canvasHeight = ...clientHeight;
```

When page is loaded with browser zoom  125% the dashed border is reduced (if zoom is higher than 100%) and Localizer lines is absent at all:
![Zoom: 125%](https://user-images.githubusercontent.com/26387367/37016474-716a0560-211d-11e8-812c-1338753863de.png)
The larger the initial zoom the smaller the area of the dashed border

When page is loaded with browser zoom 100% everything is normal:
![Zoom: 100%](https://user-images.githubusercontent.com/26387367/37016792-0d8fdcc0-211f-11e8-8a7f-d1824dcaf76c.png)

Tested with AMI: v0.0.23dev

Example [viewers_quadview](https://fnndsc.github.io/ami/#viewers_quadview) works perfectly with any zoom so the problem is likely in my code.

Any suggestions where to search?

**Comment by NicolasRannou** [comment]
You may want to look at the windowResize2D function, an at those lines in particular: https://github.com/FNNDSC/ami/blob/dev/examples/viewers_quadview/viewers_quadview.js#L714-L717

**Comment by NicolasRannou** [comment]
Each size the size of the renderer changes, we need to pass the new dimensions to the shaders in order to properly display those lines.

**Comment by JSibir** [comment]
Of course I have function similar to windowResize2D with those lines.
On window resize the behavior stay the same (proper on 100% browser zoom and broken on other zoom levels) as on initial page load.

I think windowResize2D function has no affect on initial page load. And my problem exist on initial page load.

**Comment by NicolasRannou** [comment]
What happens if you zoom to 50% then resize? does the resizing fix the appearance?

**Comment by JSibir** [comment]
No, behavior don't changes. Zooming and resizing after initial load changes only size and proportions, but Helpers lines works normal only if initial zoom was 100%.

**Comment by JSibir** [comment]
This is the root of all evil:

// Sets device pixel ratio. This is usually used for HiDPI device to prevent bluring output canvas.
**renderer.setPixelRatio(window.devicePixelRatio);**

Because we get something like this:
<canvas width="600" height="400" style="width: 300px; height: 200px;"></canvas>

StackHelper and LocalizerHelper doesn't support this feature.

**Comment by eduardor2k** [comment]
Hi, I have the same problem, did you find a solution for this problem?

**Comment by JSibir** [comment]
For now `setPixelRatio` break `stackHelper` borders and `localizerHelper` lines, so I don't use `setPixelRatio` and everything work fine.
